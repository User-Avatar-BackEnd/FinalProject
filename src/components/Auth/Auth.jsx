import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import Login from './Login/Login';
import Registration from './Registration/Registration';

import styles from './Auth.module.scss';

const Auth = ({ type }) => {
  const history = useHistory()

  const token = localStorage.getItem('AUTH_TOKEN')

  useEffect(() => {
    if (token) {
      history.replace('/')
    }
  })

  const onLogin = (token) => {
    localStorage.setItem('AUTH_TOKEN', token)
    history.push('/')
  }

  return (
    <div className={styles.Auth}>
      {type === 'login'
        ? <Login onLogin={onLogin} />
        : <Registration onLogin={onLogin} />
      }
    </div>
  );

}

export default Auth;