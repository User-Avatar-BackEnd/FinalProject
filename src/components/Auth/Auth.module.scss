.Auth {
  width: 400px;
  margin: 50px auto;
  background-color: aliceblue;
  min-height: 400px;
  border-radius: 10px;
  padding: 20px 0;
  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
}