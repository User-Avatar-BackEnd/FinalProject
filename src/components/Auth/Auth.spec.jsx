import React from 'react';
import { shallow } from 'enzyme';

import Auth from './Auth';

describe('AdminEvents component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Auth />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should match snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });
});