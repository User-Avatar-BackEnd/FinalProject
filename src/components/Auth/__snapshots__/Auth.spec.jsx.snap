// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRank component should match snapshot 1`] = `
<div
  className="UserRank"
>
  <Registration
    onLogin={[Function]}
  />
</div>
`;
